sda-download/api/sda/sda.go

```diff
@@ -268,7 +313,24 @@ func Download(c *gin.Context) {
 
 	switch c.Param("type") {
 	case "encrypted":
-		fileStream = encryptedFileReader
+		// The key provided in the header should be base64 encoded
+		reencKey := c.GetHeader("Client-Public-Key")
+		if strings.HasPrefix(c.GetHeader("User-Agent"), "htsget") {
+			reencKey = c.GetHeader("Server-Public-Key")
+		}
```

Comment:
Can't we just define the order so that `Server-Public-Key` overrides `Client-Public-Key` rather than looking at `User-Agent`?

Reply:
(I left the previous comment, in case anyone read it)
On a second thought, tools like samtools might do the whole procedure in one request:
- Make the request to htsget and get the results
- Make the requests to sda-download and get the relevant parts of the file
- Concatenate everything together
If that's the workflow, then the `Client-Public-Key` needs to be sent with the initial request. This is only an assumption currently, as samtools doesn't support custom header, but I think we should leave it as is for now and possibly update it when we have more information.

Reply:
Ok, I'm not thrilled with having key being picked based on `User-Agent` but am fine with leaving as is for now.